extensions/web-dependency-locator/runtime/src/main/java/io/quarkus/webdependency/locator/runtime/WebDependencyLocatorRecorder.java

```diff
@@ -11,30 +13,38 @@
             Map<String, String> webDependencyNameToVersionMap) {
         return (event) -> {
             String path = event.normalizedPath();
             if (path.startsWith(webDependenciesRootUrl)) {
-                String rest = path.substring(webDependenciesRootUrl.length());
-                String webdep = rest.substring(0, rest.indexOf('/'));
-                if (webDependencyNameToVersionMap.containsKey(webdep)) {
-                    // Check this is not the actual path (ex: /webjars/jquery/${jquery.version}/...
-                    int endOfVersion = rest.indexOf('/', rest.indexOf('/') + 1);
-                    if (endOfVersion == -1) {
-                        endOfVersion = rest.length();
-                    }
-                    String nextPathEntry = rest.substring(rest.indexOf('/') + 1, endOfVersion);
-                    if (webDependencyNameToVersionMap.get(webdep) == null
-                            || nextPathEntry.equals(webDependencyNameToVersionMap.get(webdep))) {
-                        // go to the next handler (which should be the static resource handler, if one exists)
-                        event.next();
+                try {
+                    String rest = path.substring(webDependenciesRootUrl.length());
+                    String webdep = rest.substring(0, rest.indexOf('/'));
+                    if (webDependencyNameToVersionMap.containsKey(webdep)) {
+                        // Check this is not the actual path (ex: /webjars/jquery/${jquery.version}/...
+                        int endOfVersion = rest.indexOf('/', rest.indexOf('/') + 1);
+                        if (endOfVersion == -1) {
+                            endOfVersion = rest.length();
+                        }
+                        String nextPathEntry = rest.substring(rest.indexOf('/') + 1, endOfVersion);
+                        if (webDependencyNameToVersionMap.get(webdep) == null
+                                || nextPathEntry.equals(webDependencyNameToVersionMap.get(webdep))) {
+                            // go to the next handler (which should be the static resource handler, if one exists)
+                            event.next();
+                        } else {
+                            // reroute to the real resource
+                            event.reroute(webDependenciesRootUrl + webdep + "/"
+                                    + webDependencyNameToVersionMap.get(webdep) + rest.substring(rest.indexOf('/')));
+                        }
                     } else {
-                        // reroute to the real resource
-                        event.reroute(webDependenciesRootUrl + webdep + "/"
-                                + webDependencyNameToVersionMap.get(webdep) + rest.substring(rest.indexOf('/')));
+                        event.next();
                     }
-                } else {
+                } catch (Throwable t) {
+                    LOG.error("Error while locating web jar " + path);
```

Comment:
Do we really want to log an error? You will have a 404 on the client side and I think that's enough. I wouldn't want to have my logs filled if someone triggers a 404.

Reply:
Agree. I'll remove.